db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1627 @@
+
+        // we expect only 1 result
+        assertTrue(res.next());
+
+        // res.getArray() doesn't give you a String[] array - have to do some manipulation to get its length as a result
+        System.out.println(res.getArray("sources").toString());
+        assertEquals(4, res.getArray("sources").toString().split("\",\"").length);
+        assertEquals(7, res.getInt("derived_total"));
+
+        // we expect only 1 result
+        assertFalse(res.next());
+      });
+    }
+
+    /**
+     * This test uploads data (tests 01-04 from SSMO-MPS/mission-data-sandbox/derivation_test) and then confirms
+     * deletion works as expected. It is not a rigorous test, as we only focus on one order (any other order is
+     * logically equivalent to testing constraints, which is guaranteed by PSQL and tested later). We follow the
+     * correct order of:
+     * - derivation_group_comp (view)
+     * - derived_events (view)
+     * - subtract_later_ranges (function)
+     * - check_external_event_boundaries (trigger, then function)
+     * - external_source_pdg_association (trigger, then function)
+     * - plan_derivation_group
+     * - check_external_event_duration_is_nonnegative_trigger (trigger)
+     * - external_event
+     * - external_event_type
+     * - external_source
+     * - derivation_group
+     * - external_source_type
+     * though it is possible to rearrange this order, so long as views are deleted before functions, functions are
+     * deleted before tables, events are deleted before their types, sources deleted before their types but after
+     * events, and derivation groups deleted after linked sources removed but before source types removed, things work
+     * as expected.
+     *
+     * For completeness, another arbitrary ordering (source type deleted before derivation group) is included as a basic
+     * example of failure. A lot is possible though; deleting functions before triggers, tables before views, etc.
+     */
+    @Test
+    void verifyDeletion() {
+      final var statement = getStatement();
+
+      // correct deletion order (a small amount of variance is allowed here, see documentation of this function).
+      assertDoesNotThrow(() -> merlinHelper.upload_source(dg));
+      assertDoesNotThrow(() -> {
+        statement.get().executeUpdate(
+            // language=sql
+            """
+            DROP VIEW ui.derivation_group_comp;
+            DROP VIEW merlin.derived_events;
+            DROP FUNCTION merlin.subtract_later_ranges;
+            DROP TRIGGER check_external_event_boundaries ON merlin.external_event;
+            DROP FUNCTION merlin.check_external_event_boundaries;
+            DROP TRIGGER external_source_pdg_association_delete ON merlin.external_source;
+            DROP FUNCTION merlin.external_source_pdg_association_delete;
+            DROP TABLE merlin.plan_derivation_group;
+            DROP TRIGGER check_external_event_duration_is_nonnegative_trigger ON merlin.external_event;
```

Comment:
CRITICAL: DB tests should **NEVER** change the DB schema. Fix this or drop the test. This issue happens again beginning on L246.